versions.tf

```diff
@@ -4,7 +4,7 @@ terraform {
   required_providers {
     aws = {
       source  = "hashicorp/aws"
-      version = ">= 3.35.0"
+      version = ">= 5.15.0"
```

Comment:
If we're bumping to this version, this is considered a major version bump as it requires users to bump to a new provider version. Just calling that out explicitly here since this is now a larger change than just a new var + argument usage.  